Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -192,14 +192,36 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
     cookieJar.setCookieAsync(document, url, cookie, WTFMove(completionHandler));
 }
 
-void CookieStore::remove(const String&, Ref<DeferredPromise>&& promise)
+void CookieStore::remove(String&& name, Ref<DeferredPromise>&& promise)
 {
-    promise->reject(NotSupportedError);
+    remove(CookieStoreDeleteOptions { WTFMove(name), { } }, WTFMove(promise));
 }
 
-void CookieStore::remove(CookieStoreDeleteOptions&&, Ref<DeferredPromise>&& promise)
+void CookieStore::remove(CookieStoreDeleteOptions&& options, Ref<DeferredPromise>&& promise)
 {
-    promise->reject(NotSupportedError);
+    auto* context = scriptExecutionContext();
+    if (!context) {
+        promise->reject(SecurityError);
+        return;
+    }
+
+    auto* origin = context->securityOrigin();
+    if (!origin) {
+        promise->reject(SecurityError);
+        return;
+    }
+
+    if (origin->isOpaque()) {
+        promise->reject(Exception { SecurityError, "The origin is opaque"_s });
+        return;
+    }
+
+    CookieInit initOptions;
```

Comment:
This should forward the CookieStoreDeleteOptions's domain/path to initOptions too.

Reply:
That's what I had initially, but I excluded them because the spec says to use `null` and `'/'` for the domain and path when running delete (https://wicg.github.io/cookie-store/#CookieStore-delete). So I just let domain and path be default initialized to these values. Should I change this?

Reply:
I see this in the spec:
```
Let r be the result of running [delete a cookie](https://wicg.github.io/cookie-store/#delete-a-cookie) with url, options["[name](https://wicg.github.io/cookie-store/#dom-cookiestoredeleteoptions-name)"], options["[domain](https://wicg.github.io/cookie-store/#dom-cookiestoredeleteoptions-domain)"], and options["[path](https://wicg.github.io/cookie-store/#dom-cookiestoredeleteoptions-path)"].
```

Reply:
`null` and `/` are only for the delete() that takes a name, not for the delete that takes options as far as I can tell.

Reply:
Ok that makes sense.